routes/project/inference_api_key.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Clover
+  hash_branch(:project_prefix, "inference-api-key") do |r|
+    r.web do
+      r.get true do
+        @inference_api_keys = Serializers::InferenceApiKey.serialize(inference_api_key_ds.all)
+        view "inference/api_key/index"
+      end
+
+      r.post true do
+        authorize("InferenceApiKey:create", @project.id)
+        it = DB.transaction { ApiKey.create_inference_api_key(@project) }
+        flash["notice"] = "Created Inference API Key with id #{it.ubid}"
+        r.redirect "#{@project.path}/inference-api-key"
+      end
+
+      r.delete String do |ubid|
+        if (it = ApiKey.from_ubid(ubid))
+          authorize("InferenceApiKey:delete", it.id)
```

Comment:
Not an issue with this change, but this technically leaks information about whether a ubid represents a valid ApiKey, even if the ApiKey is associated to a different project.  Not sure that matters, and Clover may leak the same information elsewhere.  To avoid this case leaking, you could switch to `it = inference_api_key_ds.with_pk(UBID.to_uuid(ubid))`.

Reply:
thanks for the catch. applied your suggestion.